File: libgomp/testsuite/libgomp.oacc-fortran/data-already-3.f
! { dg-skip-if "" { *-*-* } { "*" } { "-DACC_MEM_SHARED=0" } }

      IMPLICIT NONE
      INCLUDE "openacc_lib.h"

      INTEGER I

!$ACC DATA PRESENT_OR_COPY (I)
      WRITE(0, *) "CheCKpOInT"
      CALL ACC_COPYIN (I)
!$ACC END DATA

      END

! { dg-output "CheCKpOInT(\n|\r\n|\r).*" }
! { dg-output "already mapped to" }
! { dg-shouldfail "" }

File: libgomp/testsuite/libgomp.oacc-c-c++-common/data-already-3.c
/* { dg-skip-if "" { *-*-* } { "*" } { "-DACC_MEM_SHARED=0" } } */

#include <stdio.h>
#include <openacc.h>

int
main (int argc, char *argv[])
{
  int i;

#pragma acc data present_or_copy (i)
  {
    fprintf (stderr, "CheCKpOInT\n");
    acc_copyin (&i, sizeof i);
  }

  return 0;
}

/* { dg-output "CheCKpOInT(\n|\r\n|\r).*" } */
